test/python/test_rewrites_ast.py

```diff
@@ -212,3 +228,84 @@ def test_lifting_over_build(prelude_symtab):
     match3 = utils.single_elem(list(lift_if_matcher.find_all_matches(e3)))
     actual3 = match3.apply_rewrite()
     assert actual3 == expected3
+
+
+def test_new_bind(prelude_symtab):
+    symtab = {**prelude_symtab, "x": Type.Integer}
+    input = parse_expr_string("(let (y (add x 1)) (add y (add x 1)))")
+    expected = parse_expr_string("(let (y (add x 1)) (add y (let (z (add x 1)) z)))")
+    expected_raw = [expected] + [
+        parse_expr_string("(let (z (let (y (add x 1)) (add y (add x 1)))) z)"),
+        parse_expr_string("(let (y (let (z (add x 1)) z)) (add y (add x 1)))"),
+        parse_expr_string("(let (y (add x 1)) (let (z (add y (add x 1))) z))"),
+    ]
+    type_propagate_decls(expected_raw + [input], symtab)
+    actual_raw = [
+        m.apply_rewrite() for m in RuleSet([raw_new_bind]).find_all_matches(input)
+    ]
```

Comment:
This pattern is sufficiently common, I feel like `apply_all_matches` might be a good utility function.  
Then this line would read
```python
actual_raw = apply_all_rewrites(RuleSet([raw_new_bind]), input)
```

Reply:
Or `apply_all_matching_rewrites` but it feels redundant as one can't apply a non-matching rewrite...

Reply:
Yes, totally. Do you have a preference for where to put test utility functions? `test/.../testutils.py` or somewhere in src/ e.g. `src/python/ksc/rewrites.py` ?

Reply:
For now I've just made a local function common to this file, but could consider commoning up with `sorted_rewrites` which a couple of other test files have their own definitions (plural!) of.